x-pack/plugins/security_solution/public/common/components/event_details/insights/related_alerts_by_process_ancestry.tsx

```diff
@@ -167,18 +168,57 @@ const ActualRelatedAlertsByProcessAncestry: React.FC<{
   eventId: string;
   timelineId?: string;
 }> = ({ alertIds, eventId, timelineId }) => {
+  const dataProviderLimit = 5;
   const dataProviders = useMemo(() => {
     if (alertIds && alertIds.length) {
-      return alertIds.reduce<DataProvider[]>((result, alertId, index) => {
-        const id = `${timelineId}-${eventId}-event.id-${index}-${alertId}`;
-        result.push(getDataProvider('_id', id, alertId));
-        return result;
-      }, []);
+      if (alertIds.length > dataProviderLimit) {
+        return null;
+      } else {
+        return alertIds.reduce<DataProvider[]>((result, alertId, index) => {
+          const id = `${timelineId}-${eventId}-event.id-${index}-${alertId}`;
+          result.push(getDataProvider('_id', id, alertId));
+          return result;
+        }, []);
+      }
     }
     return null;
-  }, [alertIds, eventId, timelineId]);
+  }, [alertIds, eventId, timelineId, dataProviderLimit]);
+
+  const filters: Filter[] | null = useMemo(() => {
```

Comment:
Do you think below function can simply be reused here? Also shouldn't below query has `filter` clause as well?

https://github.com/elastic/kibana/blob/61630e32fd514c9ed7ee0dfa10c3abc8684f9628/x-pack/plugins/security_solution/public/detections/components/alerts_table/actions.tsx#L285-L289